src/ui/screens/AddBountyScreen.tsx

```diff
@@ -0,0 +1,160 @@
+import {useSnackbar} from 'context/SnackbarContext';
+import {InputLabel} from '@ui/library/InputLabel';
+import {decimalKeypad} from 'src/utils';
+import {useFormatBalance} from 'src/api/hooks/useFormatBalance';
+import {useBountiesSummary} from 'src/api/hooks/useBountiesSummary';
+import type {BN} from '@polkadot/util';
+import {formattedStringToBn} from 'src/api/utils/balance';
+
+export function AddBountyScreen({navigation}: {navigation: NavigationProp<AppStackParamList>}) {
+  const {colors} = useTheme();
+  const modalRef = useRef<Modalize>(null);
+  useEffect(() => {
+    modalRef.current?.open();
+  }, []);
+
+  const {data: bounty} = useBountiesSummary();
+  const {formatBalance, stringToBn} = useFormatBalance();
+
+  const [bountyTitle, setBountyTitle] = React.useState('');
+  const [bountyAllocation, SetBountyAllocation] = React.useState('');
+
+  const [account, setAccount] = React.useState<Account>();
+  const modelClose = () => modalRef.current?.close();
+  const startTx = useApiTx();
+  const {api} = useApi();
+  const snackbar = useSnackbar();
+
+  const {calculatedBountyBond, bountyAllocationBN, isBountyValid, isBountyTitleValid} = useMemo(() => {
+    if (bounty && bounty.bountyDepositBase) {
+      const calculatedBountyBond = calculateBountyBond(
+        bountyTitle,
+        formattedStringToBn(bounty.bountyDepositBase),
+        formattedStringToBn(bounty.dataDepositPerByte),
+      );
+      const bountyAllocationBN = stringToBn(bountyAllocation);
+      const isBountyValid = bountyAllocationBN
+        ? formattedStringToBn(bounty.bountyValueMinimum).gt(bountyAllocationBN)
+        : false;
+      const isBountyTitleValid = Number(bounty?.maximumReasonLength) <= Number(bountyTitle.length);
+      return {calculatedBountyBond: calculatedBountyBond, bountyAllocationBN, isBountyValid, isBountyTitleValid};
+    }
+    return {
+      calculatedBountyBond: formattedStringToBn(bounty?.bountyDepositBase),
+      bountyAllocationBN: stringToBn(bountyAllocation),
+      isBountyValid: false,
+      isBountyTitleValid: bountyTitle.length < 1,
+    };
+  }, [bountyTitle, bountyAllocation, bounty, stringToBn]);
+
+  const disabled = !bountyAllocation || !account || isBountyValid || isBountyTitleValid;
+
+  const submitBounty = () => {
+    if (account && api && bountyAllocationBN) {
+      startTx({
+        address: account.address,
+        txMethod: `${api.tx.bounties ? `bounties.proposeBounty` : `treasury.proposeBounty`}`,
+        params: [bountyAllocationBN, bountyTitle],
+      })
+        .then(() => {
+          snackbar('New bounty created');
+        })
+        .catch(() => {
+          snackbar('Error while submitting bounty');
+        });
+      modalRef.current?.close();
+    }
+  };
+
+  return (
+    <Modalize ref={modalRef} adjustToContentHeight onClose={navigation.goBack} panGestureEnabled={false}>
+      <Layout style={styles.container}>
+        <View style={styles.header}>
```

Comment:
```suggestion
        <View style={globalStyles.alignCenter}>
```

Also please clean-up if some styles are not needed anymore.